### PR TITLE
1327 location picker

### PR DESCRIPTION
**Pull request by Angamanga**
**Ready for review**

**This pull request makes the following changes:**
- Adding location-picker in accordance with the pattern library

**Test these changes by:**
- Go to map, choose add new post
- Look at the map, try writing a location in the input-box
- Check the behaviour of the dropdown
- Try clicking search and choose a result
- Try using current location
- Compare to pattern library : http://preview.ushahidi.com/platform-pattern-library/master/assets/html/5_layouts/post-edit.html

**Things to consider when QAing:**
- The level of detail in the results is very high, and these are the results from the geocoding-api we are using. Is this ok enough or does it need to be tweaked?
- Check the behaviour of the dropdown. Is it ok/logical to the user? Would it be more logical/userfriendly to have a 'close'-button too since its not auto-closing when clicking outside?

Fixes ushahidi/platform#1327 .

Ping @ushahidi/platform

---
This change is [Reviewable](https://reviewable.io/reviews/ushahidi/platform-client/494)

### CODE REVIEW

File: app/main/posts/modify/location.html [hunk 1]
```diff
@@ -1,15 +1,55 @@
...
-            <input name="{{name}}" type="text" placeholder="{{'location.placeholder' | translate}}" ng-model="searchLocationTerm" ng-disabled="processing">
-            <button class="btn btn-info" type="submit" ng-disabled="processing">
-                <div class="loading" ng-show="processing"><div class="line"></div><div class="line"></div><div class="line"></div></div>
-                <span translate>location.search<span>
-            </button>
-
-            <button class="button-secondary alt" type="button" ng-click="clear()" ng-disabled="processing" translate>location.clear</button>
+ 
+<div class="location-wrapper">
+    <div id="map" class="map" style="height: 265px; width: 100%;"></div>
```
Comment (rjmackay) [inline app/main/posts/modify/location.html]: `id` here still needs to be unique. We can have multiple location editors on a page.

File: app/main/posts/modify/location.directive.js [hunk 2]
```diff
@@ -39,22 +47,20 @@ function PostLocationDirective($http, L, Geocoding, Maps, _, Notify, $window) {
...
                 }
-
                 map.on('click', onMapClick);
                 // treate locationfound same as map click
                 map.on('locationfound', onMapClick);
-
                 // Add locate control, but only on https
                 if (window.location.protocol === 'https:' || window.location.hostname === 'localhost') {
-                    L.control.locate({
+                    $scope.currentPositionControl = L.control.locate({
```
Comment (rjmackay) [inline app/main/posts/modify/location.directive.js]: Does this need to be bound to $scope? Could we just use a local variable?

File: app/main/posts/modify/location.html (hunk 1 above)
Comment (Angamanga) [inline app/main/posts/modify/location.html]: Fixed

File: app/main/posts/modify/location.directive.js (hunk 2 above)
Comment (Angamanga) [inline app/main/posts/modify/location.directive.js]: Fixed

**Comment by Angamanga** [comment]
@rjmackay Whats the process after approved review, do I go ahead merging or should I wait for you to do it?

**Pull request merged**